### ISSUE TITLE
Follow Guice best practices

### ISSUE THREAD

**Issue by deathbeam** [runelite/runelite#617]
Currently, the model we are creating when using Guice and it's injectors is not really following Guice best practices, mainly:

* [Minimize mutability](https://github.com/google/guice/wiki/MinimizeMutability)
* [Inject only direct dependecies](https://github.com/google/guice/wiki/InjectOnlyDirectDependencies)
* [Keep constructors hidden](https://github.com/google/guice/wiki/KeepConstructorsHidden)

For first point, changing all the injections from `@Inject` annotation on fields to `@Inject` annotations
on constructor should be sufficient.

For second point, f.e. ClientUI should somehow provide the plugin panel, then `net.runelite.api.Client` needs to be split to multiple interfaces and each of them needs to be provided separately, and plugins should use only the interface they really need (for example to get camera values, they should really just inject something like ClientCamera, instead of injecting Client that contains everything).

For third point, just making the constructors private should be good enough (depends on first point).

If anyone have any more ideas on how to improve the current Guice model, feel free to post them here.

**Comment by Desetude** [comment]
I think the best way to improve the current Guice model is by seperating Plugin into Plugin and Module, where the module instantiates the plugin and provides the config, etc..
This new Plugin structure will allow constructor injection, rather than field, which is evidently favourable, and the structure will generally be more flexible, e.g. with how a plugin is instantiated. Also, this new model will be a lot less hacky to implement (https://github.com/runelite/runelite/blob/master/runelite-client/src/main/java/net/runelite/client/plugins/PluginManager.java#L403 is definitely not 'natural' guice injection).

**Comment by NathanielPather** [comment]
This is still despite a solution being merged.

**Comment by deathbeam** [comment]
No, there are still quite a few cases in source where people are ignoring best practices, mainly for 1st and 3rd point.

**Comment by Nightfirecat** [comment]
(1) was addressed in 7ceb3c9e020aa8316ddc5809a2d409d5d2efb697, and (2) is as far as I'm aware, not an issue in this codebase. That means (3) should be the only point needing to be addressed, which I plan to open a PR to fix shortly.
